### PR TITLE
[CD] Add triton xpu wheel build

### PR DESCRIPTION
**Pull request by chuanqi129**
Enable triton xpu wheel build firstly, then add pytorch xpu nightly wheel build, works for #114850

### CODE REVIEW

**Comment by chuanqi129** [comment]
depends on triton xpu issue https://github.com/intel/intel-xpu-backend-for-triton/issues/1505

**Review by atalman — APPROVED** [review]
lgtm

File: .github/workflows/build-triton-wheel.yml [hunk 1]
```diff
@@ -102,19 +104,20 @@ jobs:
...
             RELEASE="--release"
           fi
 
           docker exec -t "${container_name}" yum install -y zlib-devel zip
           docker exec -t "${container_name}" "${PYTHON_EXECUTABLE}"  -m pip install -U setuptools==67.4.0
-          docker exec -t "${container_name}" "${PYTHON_EXECUTABLE}" /pytorch/.github/scripts/build_triton_wheel.py $BUILD_ROCM $RELEASE
+          # Triton xpu build use GCC11
+          if [[ "${BUILD_DEVICE}" == xpu ]]; then
+            docker exec -t "${container_name}" yum install -y devtoolset-11-gcc-c++
+            docker exec -t "${container_name}" bash -c "source /opt/rh/devtoolset-11/enable && ${PYTHON_EXECUTABLE} /pytorch/.github/scripts/build_triton_wheel.py --device=$BUILD_DEVICE $RELEASE"
```
Comment (atalman) [inline .github/workflows/build-triton-wheel.yml:117]: nit: Can we do something like this ?
```
if [[ "${BUILD_DEVICE}" == xpu ]]; then
            docker exec -t "${container_name}" yum install -y devtoolset-11-gcc-c++
            docker exec -t "${container_name}" bash -c "source /opt/rh/devtoolset-11/enable
fi
docker exec -t "${container_name}" bash -c "${PYTHON_EXECUTABLE} /pytorch/.github/scripts/build_triton_wheel.py --device=$BUILD_DEVICE $RELEASE"
```

**Pull request closed without merging**